diff -ru a/groups/bsl/bslalg/bslalg.gyp b/groups/bsl/bslalg/bslalg.gyp
--- a/groups/bsl/bslalg/bslalg.gyp
+++ b/groups/bsl/bslalg/bslalg.gyp
@@ -243,6 +243,11 @@
                         '<@(bslalg_pkgdeps)', 'bslalg' ],
       'include_dirs': [ '.' ],
       'sources': [ 'bslalg_constructorproxy.t.cpp' ],
+      # required for bslalg_constructorproxy.t linked against bsls .o's
+      # XXX: track down what results in this requirement (likely in bsls)
+      'conditions': [
+        [ 'compiler_tag == "xlC"', { 'cflags_cc': [ '-qalias=noansi' ] } ],
+      ],
     },
     {
       'target_name': 'bslalg_containerbase.t',
diff -ru a/groups/bsl/bsl+bslhdrs/bsl+bslhdrs.gyp b/groups/bsl/bsl+bslhdrs/bsl+bslhdrs.gyp
--- a/groups/bsl/bsl+bslhdrs/bsl+bslhdrs.gyp
+++ b/groups/bsl/bsl+bslhdrs/bsl+bslhdrs.gyp
@@ -1,7 +1,6 @@
 {
   'variables': {
     'bsl+bslhdrs_sources': [
-      'bslhdrs_dummy.cpp',
     ],
     'bsl+bslhdrs_tests': [
 
@@ -51,7 +50,7 @@
     },
     {
       'target_name': 'bsl+bslhdrs',
-      'type': '<(library)',
+      'type': 'none',
       'dependencies': [ '../bsl_deps.gyp:bsl_grpdeps',
                         '<@(bsl+bslhdrs_pkgdeps)',
                         'bsl+bslhdrs_sources', ],
diff -ru a/groups/bsl/bsl_deps.gyp b/groups/bsl/bsl_deps.gyp
--- a/groups/bsl/bsl_deps.gyp
+++ b/groups/bsl/bsl_deps.gyp
@@ -6,6 +6,10 @@
       'dependencies': [
 
       ],
+      # XXX: when generating .gyp from bsl metadata, need to special-case
+      'conditions': [  # (bsls_timeutil uses POSIX realtime API clock_gettime())
+        [ 'OS == "linux"', { 'link_settings': { 'libraries': [ '-lrt' ] } } ],
+      ],
     },
   ],
 }
diff -ru a/groups/bsl/bsl.gyp b/groups/bsl/bsl.gyp
--- a/groups/bsl/bsl.gyp
+++ b/groups/bsl/bsl.gyp
@@ -336,8 +336,17 @@
               },
             ],
             'all_dependent_settings': {
-                'include_dirs': [ '<(product_rollup_path)/include' ]
+                'include_dirs': [ '<(product_rollup_path)/include' ],
+                # XXX: track down source of requirement (see immediately below)
+                'conditions': [
+                  ['compiler_tag=="xlC"',{ 'cflags_cc': [ '-qalias=noansi' ] }],
+                ],
             },
+            # required for bslalg_constructorproxy.t linked against bsls .o's
+            # XXX: track down what results in this requirement (likely in bsls)
+            'conditions': [
+              [ 'compiler_tag == "xlC"', { 'cflags_cc': [ '-qalias=noansi' ] }],
+            ],
         },
         {
             'target_name': 'bsl_tests_build',
diff -ru a/groups/bsl/bslma/bslma.gyp b/groups/bsl/bslma/bslma.gyp
--- a/groups/bsl/bslma/bslma.gyp
+++ b/groups/bsl/bslma/bslma.gyp
@@ -223,6 +223,10 @@
                         '<@(bslma_pkgdeps)', 'bslma' ],
       'include_dirs': [ '.' ],
       'sources': [ 'bslma_mallocfreeallocator.t.cpp' ],
+      'conditions': [
+        [ 'OS == "aix" and library == "shared_library"',
+          { 'ldflags': [ '-bexpfull' ] } ],
+      ],
     },
     {
       'target_name': 'bslma_newdeleteallocator.t',
@@ -231,6 +235,10 @@
                         '<@(bslma_pkgdeps)', 'bslma' ],
       'include_dirs': [ '.' ],
       'sources': [ 'bslma_newdeleteallocator.t.cpp' ],
+      'conditions': [
+        [ 'OS == "aix" and library == "shared_library"',
+          { 'ldflags': [ '-bexpfull' ] } ],
+      ],
     },
     {
       'target_name': 'bslma_rawdeleterguard.t',
@@ -255,6 +263,10 @@
                         '<@(bslma_pkgdeps)', 'bslma' ],
       'include_dirs': [ '.' ],
       'sources': [ 'bslma_testallocator.t.cpp' ],
+      'conditions': [
+        [ 'OS == "aix" and library == "shared_library"',
+          { 'ldflags': [ '-bexpfull' ] } ],
+      ],
     },
     {
       'target_name': 'bslma_testallocatorexception.t',
diff -ru a/groups/bsl/bsls/bsls.gyp b/groups/bsl/bsls/bsls.gyp
--- a/groups/bsl/bsls/bsls.gyp
+++ b/groups/bsl/bsls/bsls.gyp
@@ -188,6 +188,11 @@
       # Mac OS X empty LD_DYLIB_INSTALL_NAME causes executable and shared
       # libraries linking against dylib to store same path for use at runtime
       'xcode_settings': { 'LD_DYLIB_INSTALL_NAME': '' },
+      # required for bslalg_constructorproxy.t linked against bsls .o's
+      # XXX: track down what results in this requirement (likely in bsls)
+      'conditions': [
+        [ 'compiler_tag == "xlC"', { 'cflags_cc': [ '-qalias=noansi' ] } ],
+      ],
     },
 
     {
@@ -421,6 +426,11 @@
                         '<@(bsls_pkgdeps)', 'bsls' ],
       'include_dirs': [ '.' ],
       'sources': [ 'bsls_byteorder.t.cpp' ],
+      # XXX: when generating .gyp from bsl metadata, need to special-case
+      'conditions': [
+        # ws2_32.lib required only to portably test htonl ntohl htons ntohs
+        [ 'OS == "win"',  {'link_settings': {'libraries': [ 'ws2_32.lib'] } } ],
+      ],
     },
     {
       'target_name': 'bsls_compilerfeatures.t',
diff -ru a/groups/bsl/bsl+stdhdrs/bsl+stdhdrs.gyp b/groups/bsl/bsl+stdhdrs/bsl+stdhdrs.gyp
--- a/groups/bsl/bsl+stdhdrs/bsl+stdhdrs.gyp
+++ b/groups/bsl/bsl+stdhdrs/bsl+stdhdrs.gyp
@@ -21,8 +21,13 @@
       'type': 'none',
       'direct_dependent_settings': {
         'sources': [ '<@(bsl+stdhdrs_sources)' ],
-        'include_dirs': [ '.' ],
       },
+      # one-off workaround bug in Sun Studio C++ requiring abs path to .SUNWCCh
+      'conditions': [
+        [ 'OS == "solaris"',
+          { 'direct_dependent_settings': { 'include_dirs': [ '<!(pwd)' ] } },
+          { 'direct_dependent_settings': { 'include_dirs': [ '.' ] } } ],
+      ],
     },
     {
       'target_name': 'bsl+stdhdrs_tests_build',
@@ -57,7 +62,12 @@
                         '<@(bsl+stdhdrs_pkgdeps)',
                         'bsl+stdhdrs_sources', ],
       'export_dependent_settings': [ '<@(bsl+stdhdrs_pkgdeps)' ],
-      'direct_dependent_settings': { 'include_dirs': [ '.' ] },
+      # one-off workaround bug in Sun Studio C++ requiring abs path to .SUNWCCh
+      'conditions': [
+        [ 'OS == "solaris"',
+          { 'direct_dependent_settings': { 'include_dirs': [ '<!(pwd)' ] } },
+          { 'direct_dependent_settings': { 'include_dirs': [ '.' ] } } ],
+      ],
       # Mac OS X empty LD_DYLIB_INSTALL_NAME causes executable and shared
       # libraries linking against dylib to store same path for use at runtime
       'xcode_settings': { 'LD_DYLIB_INSTALL_NAME': '' },
